### PR TITLE
fail fast on syntax error by strictly unmarshalling

### PR DESCRIPTION
**Pull request by cmeury**
* upgrade `gopkg.in/yaml` from `v1` to `v2` to have [`UnmarshalStrict()`](https://godoc.org/gopkg.in/yaml.v2#UnmarshalStrict) verify the YAML syntax
* use minikube `v0.25.2`, kubernetes `v1.9.0` and helm `v2.9.0` in integration tests
* [unrelated] adding simple happy path test for `HelmState.SyncReleases()`

Fixing #56

### CODE REVIEW

**Comment by cmeury** [comment]
Had a lot of trouble with the minikube setup, in the end imitated [habitat's way of doing things](https://github.com/kinvolk/habitat-operator/blob/master/.circleci/config.yml). Will squash this if approved.

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,34 +35,30 @@ jobs:
...
-            sudo minikube start --vm-driver=none
-            sudo chown -R $USER.$USER ~/.minikube
-            sudo chown -R $USER.$USER ~/.kube
-            minikube update-context
+            curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION}/bin/linux/amd64/kubectl
+            chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+            curl -Lo minikube https://storage.googleapis.com/minikube/releases/${MINIKUBE_VERSION}/minikube-linux-amd64
+            chmod +x minikube && sudo mv minikube /usr/local/bin/
+            sudo minikube config set WantReportErrorPrompt false
+            sudo -E minikube start --vm-driver=none --bootstrapper=localkube --kubernetes-version=${K8S_VERSION}
```
Comment (mumoshu) [inline .circleci/config.yml]: Ah. I heard that recent minikube started to rely on kubeadm to bootstrap k8s by default. Perhaps you reverted it to the old behavior with `--bootstrapper=localkube`.

Comment (cmeury) [reply .circleci/config.yml]: @mumoshu Yes, it's a remnant of trying to run the tests with minikube `v0.26.1`. It looks like they switched to kubeadm in 0.26.x and require `systemd` to be installed on the host for `vm-driver=none` to work. The CircleCI image we use does not come with it, so I had to use a different bootstrapper, but I could not get it to work. So I reverted back to 0.25.x, which makes the localkube setting redundant. I removed it now, good catch.

**Comment by mumoshu** [comment]
@cmeury Thank you so much for your contribution 🍻

**Pull request merged**